src/libcollections/string.rs

```diff
@@ -701,6 +701,13 @@ impl String {
     /// Violating these may cause problems like corrupting the allocator's
     /// internal datastructures.
     ///
+    /// The ownership of `ptr` is effectively transferred to the
+    /// `String` and the memory referred to by the pointer will be
+    /// deallocated when the `String` goes out of scope. Ensure that
+    /// nothing else uses the pointer after calling this
+    /// function. Calling this function multiple times with the same
+    /// pointer is an error.
```

Comment:
For similar reasons as in the comment above, it is not necessarily the case. If the `String` does not end up getting dropped for some reason (say the user calls mem::forget), the pointer stays completely valid.